CMakeLists.txt

```diff
@@ -22,6 +22,18 @@ endif ()
 
 set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake)
 
+
+# Enable a higher warning level and warnings-as-errors
+include (MongoC-Warnings)
+option (ENABLE_MORE_WARNINGS_AS_ERRORS "Enable extra warnings-as-errors in the build." ON)
+if (ENABLE_MORE_WARNINGS_AS_ERRORS)
+   mongoc_add_platform_compile_options (
+      msvc:/W3 msvc:/WX msvc:/wd4996
+      gnu-like:-Werror
```

Comment:
Moving `-Werror` into CMakeLists.txt guarded by an option seems good. It is less platform specific flags required to pass into cmake.

@eramongodb are you OK with the proposed change? I ask since I initially thought this contradicts https://github.com/mongodb/libmongocrypt/pull/224. Since this behind an option, I am in favor of this change.

Reply:
Thank you for the perspective. Having cmake scripts make few assumptions about the platform seems like a good goal.

Reply:
I switched the default to `OFF`. I was debating which default would be preferrable. I wish there was an easy way to detect whether we're being compiled for someone hacking on the project rather than a user building the project, especially because it would be nice to enable extra warnings in CI and have developers see those warnings by default on their local project rather than waiting to see them appear after a push.